src/App.tsx

```diff
@@ -1,61 +1,141 @@
 import './App.scss';
+import React, { useState } from 'react';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoWithUser } from './types';
+
+function todoWithUsers(): TodoWithUser[] {
+  return todosFromServer.map(todo => {
+    const foundUser = usersFromServer.find(user => user.id === todo.userId);
+
+    return {
+      ...todo,
+      user: foundUser,
+    };
+  });
+}
 
 export const App = () => {
+  const [users] = useState(usersFromServer);
+  const [todos, setTodos] = useState(todoWithUsers());
+
+  const [inputText, setInputText] = useState('');
+  const [submittedTitle, setSubmittedTitle] = useState(true);
+
+  const [selectedUser, setSelectedUser] = useState('0');
+  const [submittedUser, setSubmittedUser] = useState(true);
+
+  const submitHandle = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (inputText.trim() !== '') {
+      setSubmittedTitle(true);
+    } else {
+      setSubmittedTitle(false);
+    }
+
+    if (selectedUser !== '0') {
+      setSubmittedUser(true);
+    } else {
+      setSubmittedUser(false);
+    }
+
+    if (inputText.trim() !== '' && selectedUser !== '0') {
+      setTodos(listOfTodos => {
+        const sorted = [...todos].sort((a, b) => (a.id - b.id));
+        const newId = sorted[sorted.length - 1].id + 1;
+        const user = users.find(searchUser => {
+          return searchUser.id === Number(selectedUser);
+        });
+
+        setInputText('');
+        setSelectedUser('0');
```

Comment:
```suggestion
```
Don't call state hooks from inside another state hook. It's a bad practice that can lead to bugs.